runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -546,7 +562,18 @@ private Object applyUpdateUnsafe(SMRRecord record) {
             }
         }
 
-        // Now invoke the upcall
+        if (record.locator != null && record.locator.getGlobalAddress() >= syncTail) {
```

Comment:
We should throw an exception if `record.locator` is null.

Reply:
Currently, I use record.locator to determine whether one record is from the materialized log. It is a temporary hack. As you suggested previously, this check should be put somewhere else.

Reply:
move this piece of code to syncStreamUnsafe